1 Лабораторная

WorkerStudy opened the pull request:
Сделана 1 лабораторная работа

WorkerStudy committed:
1 Лабораторная

- Изображения
- Код

rvachev commented on Flutter/lib/Human.dart:
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
```
поменять нейминг: для файлов - lowercase_with_underscores, для классов - UpperCamelCase, для переменных, постоянных и методов - lowerCamelCase

rvachev commented on Flutter/lib/Human.dart:
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/material.dart';
+
+//Содержит в себе информацию о героях
+class Human {
```
преобразовать вид модели к следующему:
```
class Human {
  final String name;
  final String pathAvatar;
  final String info;
  final Color figColor;
  const Human(
      {required this.name,
      required this.pathAvatar,
      required this.info,
      this.figColor = Colors.green});
}
```

rvachev commented on Flutter/lib/ListHeroes.dart:
```diff
@@ -0,0 +1,62 @@
+//Работа со списком героев
+
+class ListHeroes extends StatelessWidget {
+  List<Human> Heroes = [];
+  ListHeroes() {
+    Heroes.add(
+        Human("Юи Хирасава", "assets/img/ui.jpg", "Not Info", Colors.green));
+    Heroes.add(Human(
+        "Цумуги Котобуки", "assets/img/mugi.jpg", "Not Info", Colors.blue));
+    Heroes.add(
+        Human("Мио Акияма", "assets/img/mio.jpg", "Not Info", Colors.orange));
+    Heroes.add(
+        Human("Рицу Тайнака", "assets/img/ritsu.jpg", "Not Info", Colors.pink));
+  }
```
лист моделей генерировать отдельно от виджета

rvachev commented on Flutter/lib/ListHeroes.dart:
```diff
@@ -0,0 +1,62 @@
+      itemBuilder: (context, index) {
+        return Padding(
+            padding: EdgeInsets.symmetric(horizontal: 15, vertical: 65),
+            child: Card(
+                elevation: 0.6,
+                clipBehavior: Clip.hardEdge,
+                shape: RoundedRectangleBorder(
+                    borderRadius: BorderRadius.circular(20)),
+                child: Container(
+                    decoration: BoxDecoration(
+                      image: DecorationImage(
+                          image: AssetImage(Heroes[index].Path_Avatar),
+                          fit: BoxFit.fitHeight),
+                    ),
+                    child: Column(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Expanded(child: Container()),
+                          Padding(
+                              padding: EdgeInsets.symmetric(
+                                  horizontal: 15, vertical: 15),
+                              child: Text(
+                                  style: TextStyle(
+                                      color: Colors.white,
+                                      fontSize: 24,
+                                      fontWeight: FontWeight.bold),
+                                  textAlign: TextAlign.left,
+                                  Heroes[index].Name)),
+                        ]))));
```
убрать в отдельный класс

WorkerStudy committed:
Исправление ошибок

- Изменения в наименованиях
- Преобразование класса в нормальный вид
- Отдельная генерация листа героев
- Добавление виджета карта героя
